### DISCUSSION TITLE
<C-/> open terminal in (cwd) isntead of (root dir)

### DISCUSSION THREAD

**Discussion by davidarny** [LazyVim/LazyVim#2552] [Q&A] [answered]
As per documentation `<C-/>` should behave same as `<leader>fT`

<img width="578" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17799810/0ddadf9b-22e7-4084-8733-5eedcc23de05">

But i'm facing different behaviour. `<C-/>` opens terminal in `(cwd)` for some reason. I don't have any remaps or other plugins but `extra` installed.

![Screen Recording 2024-02-14 at 17 21 12](https://github.com/LazyVim/LazyVim/assets/17799810/3251b948-3968-4867-a2f5-633630dcbf84)

**Comment by dpetka2001** [comment, marked as answer]
It's because in your `i18next` directory you have a `package.json`, so when you're in the buffer `client.js` the `root_dir` will be the `i18next` directory. But then you change to another buffer `Dockerfile.webv2`, which uses a different LSP server and it returns the corresponding `root_dir` for the Docker LSP server that you use.

**Reply by davidarny** [reply]
Yeah, got that. Suppose there is no way to change this behaviour so that terminal open in root directory of nvim, right?

**Reply by dpetka2001** [reply]
The `root_dir` is whatever the command `:LazyRoot` shows you for the corresponding buffer. That varies from LSP to LSP, because different LSP servers look for different patterns to detect the `root_dir`. So, it depends on the buffer you're currently on and the LSP servers that are attached to that buffer. The `cwd` may as well be the `root_dir` depending on the LSP server that is attached. What do you mean when you say "root directory of nvim"? What would you like to happen in your case? You can try adding in your `~/.config/nvim/lua/config/options.lua` the following `vim.g.root_spec = { "cwd" }` and see if that's the behavior you want or the other alternative would be to change the `root_pattern` for the specific LSP server you'd like to detect the `root_dir` in a different way than the default.
